src/pages/Order/Order.js

```diff
@@ -1,16 +1,83 @@
 import React, { Component } from 'react';
 import CartList from './components/CartList';
-// import Favorit from './components/Favorit';
+import Favorit from './components/Favorit';
 import OrderHeader from './components/OrderHeader';
 import './Order.scss';
 
 export default class Order extends Component {
+  state = {
+    productInCart: [],
+  };
+
+  componentDidMount() {
+    const url = '/data/cart.json';
+    fetch(url)
+      .then(res => res.json())
+      .then(data =>
+        this.setState({
+          productInCart: data.cart_info,
+        })
+      );
+  }
+
+  handleCheckBox = e => {
+    const { productInCart } = this.state;
+    const arr = productInCart.map(item => {
+      if (item.name === e.target.value) {
+        return { ...item, is_checked: !item.is_checked };
+      } else {
+        return item;
+      }
+    });
+
+    this.setState({ productInCart: arr });
+  };
```

Comment:
```suggestion
  handleCheckBox = e => {
    const { productInCart } = this.state;
    
    const nextCart = productInCart.map(item => {
      if (item.name === e.target.value) {
        return { ...item, is_checked: !item.is_checked };
      }
      
      return item
    });

    this.setState({ productInCart: nextCart });
  };
```
- 의도한 대로 잘 고쳐주셨네요!
- 어떤 상태의 다음 값이 될 변수라면 막연히 arr 같은 이름이 아니라 next 든 new든 의미를 붙여 주는게 보다 가독성 측면에서 좋을 것 같습니다.

Reply:
네!